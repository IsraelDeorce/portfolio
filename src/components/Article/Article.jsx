import './Article.scss'

function Article({ file }) {
  return <div className='article' dangerouslySetInnerHTML={{ __html: file }} />
}

export default Article