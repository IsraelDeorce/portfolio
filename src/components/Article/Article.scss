.article {
  padding: 3em 1em 1em 1em;
}

@media (min-width: 992px) {
  .article {
    padding: 0.5em 1em;
  }
}
