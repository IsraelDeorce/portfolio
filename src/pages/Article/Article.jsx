import { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { Error } from '../';
import articleContent from '../../utils/aticle-content';
import axiosConfig from '../../model/service/axiosConfig';
import { CommentsList, ArticlesList, UpvotesSection } from '../../components';

function Article() {
  const { name } = useParams();
  const article = articleContent.find(article => article.name === name)

  const [articleInfo, setArticleInfo] = useState({});
  
  useEffect(() => {
    console.log("fetch")
    try {
      const fetchData = async () => {
        const response = await axiosConfig.get(`api/articles/${name}`);
        console.log(response.data)
        setArticleInfo(response.data)
      }
      fetchData();
    } catch(error) {
      console.log(error);
    }
  }, [name])

  if (!article )
    return <Error />

  const otherArticles = articleContent.filter(article => article.name !== name);

  return (
    <>
      <h1>{article.title}</h1>
      <UpvotesSection articleName={name} upvotes={articleInfo.upvotes} setArticleInfo={setArticleInfo}/>
      {article.content.map((paragraph, key) => (
        <p key={key}>{paragraph}</p>
      ))}
      <CommentsList comments={articleInfo.comments}/>
      <h3>Other Articles:</h3>
      <ArticlesList articles={otherArticles}/>
    </>
  )
}

export default Article;