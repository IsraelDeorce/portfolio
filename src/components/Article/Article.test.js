import Article from './Article'

describe('<Article/>', () => {
  afterEach(() => jest.clearAllMocks())

  describe('Snapshot', () => {
    it('renders without crashing using shallow', () => {
      expect(shallow(<Article />)).toMatchSnapshot()
    })

    it('renders without crashing using render', () => {
      expect(render(<Article />)).toMatchSnapshot()
    })

    it('renders without crashing using mount', () => {
      expect(mount(<Article />)).toMatchSnapshot()
    })
  })
})