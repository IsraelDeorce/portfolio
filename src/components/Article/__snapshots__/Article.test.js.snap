// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article/> Snapshot renders without crashing using shallow 1`] = `
<div
  className="article"
  dangerouslySetInnerHTML={
    Object {
      "__html": undefined,
    }
  }
/>
`;
